tests/unit_tests/test_channels_first_last.py

```diff
@@ -0,0 +1,75 @@
+import unittest
+
+import tensorflow as tf
+from src.models import model_name_to_class
+
+class TestCompareChannelsFirstLast(unittest.TestCase):
+  @classmethod
+  def setUpClass(cls):
+    cls.kwargs = {
+      'input_size': [224, 224, 3],
+      'changing_ids': [],
+      'num_classes': 10,
+    }
+
+    cls.list_of_models = model_name_to_class.values()
+
+    # below model do not work for upstride types
+    remove_models = [
+      # The channel_last and first values mismatch only for upstride types. 
+      # FIXME when this is prioritized.
+      "EfficientNetB0", 
+      "EfficientNetB1",
+      "EfficientNetB2",
+      "EfficientNetB3",
+      "EfficientNetB4",
+      "EfficientNetB5",
+      "EfficientNetB6",
+      "EfficientNetB7",
+      # SeparableConv2D is not supported for upstride types
+      "NASNetCIFAR",
+      "NASNetLarge",
+      "NASNetMobile",
+    ]
+
+    tmp_list_models = model_name_to_class
+    # remove models that are not supported
+    [tmp_list_models.pop(i) for i in remove_models]
+    cls.list_models_upstride = tmp_list_models.values()
+
+  def test_compare_model_params_tensorflow(self):
+    self.kwargs.update({"upstride_type": -1, "factor": 1}) 
+    for model1, model2 in zip(self.list_of_models, self.list_of_models):
+      if model1 == model2: # would be same, just a safety check
+        # switch to channels first
+        tf.keras.backend.set_image_data_format('channels_first')
+        model_NCHW = model1(**self.kwargs).build()
+        model_NCHW_params = model_NCHW.count_params()
+        # switch back to channels last
+        tf.keras.backend.set_image_data_format('channels_last')
+        model_NHWC = model2(**self.kwargs).build()
+        model_NHWC_params = model_NHWC.count_params()
+        # compare
+        print(f"Model Name    : {model1.__name__} TensorFlow")
+        print(f"Channels_last : {model_NHWC_params:,}")
+        print(f"Channels_first: {model_NCHW_params:,}") 
+        self.assertEqual(model_NHWC_params, model_NCHW_params)
```

Comment:
I think test should be verbose only if it is not passing. By default don't need to print averything

Reply:
the test takes about 2 minutes. I started printing to if the script stuck in a loop. May be we can progress bar?

Reply:
what do you mean " if the script stuck in a loop" ?

Reply:
Since building a model takes time and we need to do for tf, type1 and type 2. This would take time. User might think the test stuck in a loop as there is no print message. I have added the tqdm to show progress bar while the user can have coffee/tea :)